reconcile/openshift_resources.py

```diff
@@ -34,3 +39,29 @@ def run(
     # check for unused resources types
     # listed under `managedResourceTypes`
     ob.check_unused_resource_types(ri)
+
+
+def early_exit_desired_state(*args, **kwargs) -> dict[str, Any]:
+    settings = queries.get_secret_reader_settings()
+    namespaces, _ = orb.get_namespaces(PROVIDERS)
+    orb.lookup_secret = (
+        lambda path, key, version=None, tvars=None, settings=None: f"vault({path}, {key}, {version})"
+    )
+    orb.lookup_github_file_content = (
+        lambda repo, path, ref, tvars=None, settings=None: f"github({repo}, {path}, {ref})"
+    )
+    orb.url_makes_sense = lambda url: False
+    orb.check_alertmanager_config = (
+        lambda data, path, alertmanager_config_key, decode_base64=False: True
+    )
+    resources = []
+    for ns_info in namespaces:
+        for r in ns_info["openshiftResources"]:
+            resources.append(
+                orb.fetch_openshift_resource(r, ns_info, settings=settings).body
+            )
```

Comment:
how long does this take to execute?

Reply:
too long. working on threading.

Reply:
threading doesn't make a significant improvement. the current time is ~1.5 minutes.

Reply:
i will work on this in a follow up PR. leaving this sequential for now.

Reply:
i'm ok with keeping the code but not activate it during PR checks.

Reply:
I think the problem is not fetching but deserializing the YAML resources. That's something that won't work well with threads :(

Reply:
i agree with the direction. this is something that is already not working well in openshift-resources.

Reply:
related discussion also in #2712